spec/javascripts/modules/track-selected-taxon.spec.js

```diff
@@ -0,0 +1,145 @@
+                <div class="govuk-checkboxes__item">
+                  <input class="govuk-checkboxes__input" type="checkbox" name="miller-columns-04c58f82[]" value="childcare-and-early-years" id="miller-columns-04c58f82-0-1" tabindex="-1">
+                  <label class="govuk-label govuk-checkboxes__label" for="miller-columns-04c58f82-0-1">
+                    Childcare and early years
+                  </label>
+                </div>
+              </li>
+            </ul>
+          </div>
+          <div class="miller-columns__column miller-columns__column--active">
+            <button class="govuk-back-link" type="button">Back</button>
+            <h3 class="miller-columns__column-heading">Childcare and early years</h3>
+            <ul class="miller-columns__column-list">
+              <li class="miller-columns__item govuk-checkboxes--small" aria-describedby="navigation-instructions" tabindex="0">
+                <div class="govuk-checkboxes__item">
+                  <input class="govuk-checkboxes__input" type="checkbox" name="miller-columns-04c58f82[]" value="childcare-vouchers" id="miller-columns-04c58f82-0-1-0" tabindex="-1">
+                  <label class="govuk-label govuk-checkboxes__label" for="miller-columns-04c58f82-0-1-0">
+                    Childcare vouchers
+                  </label>
+                </div>
+              </li>
+            </ul>
+          </div>
+        </miller-columns>
+      </div>
+      <div class="govuk-button-group govuk-!-margin-top-7">
+        <button
+          class="gem-c-button govuk-button" type="submit"
+          name="save" value="save">
+          Update tags
+        </button>
+        <button
+          class="gem-c-button govuk-button govuk-button--secondary" type="submit"
+          name="legacy_tags" value="legacy_tags">
+          Update and review specialist topic tags
+        </button>
+      </div>
+    `
+
+    var trackSelectedTaxons = new GOVUK.Modules.TrackSelectedTaxons(form)
+    trackSelectedTaxons.init()
+  })
+
+  it('should send tracking events for each selected taxon when form is submitted', function () {
+    spyOn(GOVUK.analytics, 'trackEvent')
+    form.dispatchEvent(new Event('submit'))
+
+    expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith(
+      'taxonSelection',
+      'Parenting, childcare and children\'s services > Divorce, separation and legal issues',
+      { label: '/government/admin/editions/1/tags/edit' }
+    )
+
+    expect(GOVUK.analytics.trackEvent).toHaveBeenCalledWith(
+      'taxonSelection',
+      'Parenting, childcare and children\'s services > Childcare and early years',
+      { label: '/government/admin/editions/1/tags/edit' }
+    )
+  })
+
+  it('should send two tracking event when form is submitted with withdrawal date selected', function () {
```

Comment:
You've got the wrong test description here (but I can tell where you copied & pasted this test from 😆)

Reply:
updated